### ISSUE TITLE
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.

### ISSUE THREAD

**Issue by liusfore** [Rappsilber-Laboratory/AlphaLink2#7]
when i excute pip install https://github.com/dptech-corp/Uni-Core/releases/download/0.0.3/unicore-0.0.1+cu118torch2.0.0-cp310-cp310-linux_x86_64.whl on my ubuntu computer, there are many dependency conflicts which are described as 
"
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
argilla 1.9.0 requires pydantic>=1.10.7, which is not installed.
httpcore 0.16.3 requires anyio<5.0,>=3.0, which is not installed.
httpcore 0.16.3 requires h11<0.15,>=0.13, which is not installed.
httpcore 0.16.3 requires sniffio==1.*, which is not installed.
httpx 0.23.3 requires sniffio, which is not installed.
icetk 0.0.7 requires sentencepiece, which is not installed.
langchain 0.0.198 requires aiohttp<4.0.0,>=3.8.3, which is not installed.
langchain 0.0.198 requires async-timeout<5.0.0,>=4.0.0; python_version < "3.11", which is not installed.
langchain 0.0.198 requires pydantic<2,>=1, which is not installed.
langchainplus-sdk 0.0.9 requires pydantic<2,>=1, which is not installed.
layoutparser 0.3.4 requires pillow, which is not installed.
nltk 3.8.1 requires regex>=2021.8.3, which is not installed.
pdfplumber 0.9.0 requires Pillow>=9.1, which is not installed.
pytesseract 0.3.10 requires Pillow>=8.0.0, which is not installed.
sentence-transformers 2.2.2 requires huggingface-hub>=0.4.0, which is not installed.
sentence-transformers 2.2.2 requires sentencepiece, which is not installed.
sentence-transformers 2.2.2 requires transformers<5.0.0,>=4.6.0, which is not installed.
timm 0.9.2 requires huggingface-hub, which is not installed.
torchvision 0.15.2 requires pillow!=8.3.*,>=5.3.0, which is not installed.
unstructured 0.7.4 requires lxml, which is not installed.
unstructured 0.7.4 requires markdown, which is not installed.
unstructured 0.7.4 requires pillow, which is not installed.
unstructured 0.7.4 requires python-docx, which is not installed.
unstructured-inference 0.5.1 requires huggingface-hub, which is not installed.
unstructured-inference 0.5.1 requires python-multipart, which is not installed.
unstructured-inference 0.5.1 requires transformers>=4.25.1, which is not installed.
zh-langchain 0.2.0 requires beautifulsoup4, which is not installed.
zh-langchain 0.2.0 requires cpm-kernels, which is not installed.
icetk 0.0.7 requires protobuf<3.19, but you have protobuf 4.23.4 which is incompatible.
"

**Comment by liusfore** [comment]
Then, i try the following one
(alphalink) dell@dell:/media/dell/data1$ pip install pydantic>=1.10.7 
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
langchain 0.0.198 requires aiohttp<4.0.0,>=3.8.3, which is not installed.
langchain 0.0.198 requires async-timeout<5.0.0,>=4.0.0; python_version < "3.11", which is not installed.
unstructured 0.7.4 requires lxml, which is not installed.
unstructured 0.7.4 requires markdown, which is not installed.
unstructured 0.7.4 requires pillow, which is not installed.
unstructured 0.7.4 requires python-docx, which is not installed.
zh-langchain 0.2.0 requires beautifulsoup4, which is not installed.
zh-langchain 0.2.0 requires cpm-kernels, which is not installed.
langchain 0.0.198 requires pydantic<2,>=1, but you have pydantic 2.0.3 which is incompatible.
langchainplus-sdk 0.0.9 requires pydantic<2,>=1, but you have pydantic 2.0.3 which is incompatible.

**Comment by lhatsk** [comment]
AFAICT, those conflicts are from packages installed in your base environment, e.g., langchain, icetk etc. Easiest would be to set up a fresh conda environment and leave base untouched.

**Comment by liusfore** [comment]
The alphalink is a newly set conda envirnment, and the problem is still happened

**Comment by lhatsk** [comment]
Yes, but the conflicts happen because of the packages you installed into the base environment. Packages from the base environment are part of every conda environment (see e.g., langchain which is not installed by AlphaLink). It's best to avoid installing packages into the base environment to keep a complete separation. With fresh conda environment, I actually meant a fresh installation/ different path with a new base environment.
